eZ/Bundle/EzPublishCoreBundle/Imagine/Variation/AliasGeneratorDecorator.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine\Variation;
+
+use eZ\Bundle\EzPublishCoreBundle\Imagine\IORepositoryResolver;
+use eZ\Bundle\EzPublishCoreBundle\Imagine\VariationPathGenerator;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\IO\IOServiceInterface;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Exception\SourceImageNotFoundException;
+use eZ\Publish\SPI\Variation\Values\ImageVariation;
+use eZ\Publish\SPI\Variation\VariationHandler;
+use Imagine\Image\ImagineInterface;
+
+/**
+ * Alias Generator Decorator which ensures (using Imagine if needed) that ImageVariation has proper
+ * dimensions.
+ */
+class AliasGeneratorDecorator implements VariationHandler
```

Comment:
> How about renaming it to `AliasDimensionsDecorator`? IMHO it will better describe what this class is used for.

Hmm, then to me it would mean "something that decorates class returning alias dimensions". What we have here is another layer decorating AliasGenerator returning the same result. 
But I agree naming is not the best here.

How about `ImagineAwareAliasGenerator`? Imagine is hardcoded in the namespace anyway.